src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleDictionary.cs

```diff
@@ -53,6 +53,14 @@ private static void HandleStartDictionary(JsonSerializerOptions options, ref Rea
 
                     state.Current.ReturnValue = classInfo.CreateObject();
                 }
+                else if (options.ReferenceHandling.ShouldReadPreservedReferences() && state.Current.IsProcessingEnumerable())
+                {
+                    Type preservedObjType = state.Current.JsonClassInfo.PolicyProperty.GetJsonPreservedReferenceType();
```

Comment:
because every Preserved Array object needs to be of type `JsonPreservedReference<T>` where `T` is the type of the `TDeclaredProperty` of the current array property or root object.